src/Fixer/FunctionNotation/PhpdocToParamTypeFixer.php

```diff
@@ -0,0 +1,317 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\FunctionNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\DocBlock\Annotation;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\Fixer\ConfigurationDefinitionFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Jan Gantzert <jan@familie-gantzert.de>
+ */
+final class PhpdocToParamTypeFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * @var array
+     */
+    private $blacklistFuncNames = [
+        [T_STRING, '__construct'],
+        [T_STRING, '__destruct'],
+        [T_STRING, '__clone'],
+    ];
+
+    /**
+     * @var array
+     */
+    private $skippedTypes = [
+        'mixed' => true,
+        'resource' => true,
+        'null' => true,
+        'static' => true,
+    ];
+
+    /**
+     * @var array
+     */
+    private $versionSpecificTypes = [
+        'void' => 70100,
+        'iterable' => 70100,
+        'object' => 70200,
+    ];
+
+    /**
+     * @var string
+     */
+    private $classRegex = '/^\\\\?[a-zA-Z_\\x7f-\\xff](?:\\\\?[a-zA-Z0-9_\\x7f-\\xff]+)*(?<array>\[\])*$/';
```

Comment:
Is this something we have/want in a more general place? I can't imagine it's only used here, especially seeing there already is a `PhpdocToReturnTypeFixer`.

After checking, I see the exact regex is listed there, and who knows where more. This is just duplication.

Reply:
Correct, this is a duplication from PhpdocToReturnTypeFixer, but not anywhere else.
As long as there aren`t more fixers I would not create an abstract, parent or whatever class for one line.

Reply:
I'm not talking about creating a new class, I'm saying maybe there already is a class where this is more in place. Like for example, would it make sense to add it to `Annotation`? Just thinking out loud here.

Reply:
> Like for example, would it make sense to add it to Annotation?

:+1: Or even better `AnnotationRegex` or similar, to state purporse of such class. 

Also, this should be a constant, since code should not change this value.

```php
AnnotationRegex::CLASS_NAME
```

Reply:
I converted it to a local constant .... PhpdocToParamTypeFixer::CLASS_REGEX